client/state/themes/selectors/get-theme.js

```diff
@@ -20,3 +34,29 @@ export const getTheme = createSelector(
 	},
 	( state ) => state.themes.queries
 );
+
+export const isThemeInstalledOnSite = createSelector(
+	( state, sourceAndSiteId, themeId ) => {
+		const query = '[["request_type","my-themes"]]';
+		let siteId = sourceAndSiteId;
+
+		if ( 'wpcom' === sourceAndSiteId && isUserLoggedIn( state ) ) {
+			siteId = getSelectedSiteId( state );
+		} else if ( 'wpcom' === sourceAndSiteId && ! isUserLoggedIn( state ) ) {
+			siteId = 'wpcom';
+		}
```

Comment:
Same as above, we could simplify a bit, and given that this same code is duplicated we could extract it to a function to avoid duplicity and reduce maintenance costs.